launcher/sdw_updater_gui/UpdaterApp.py

```diff
@@ -158,6 +189,26 @@ def exit_launcher(self):
         sys.exit()
 
 
+def _is_netcheck_successful() -> bool:
+    """
+    Helper function to assess network connectivity before launching updater.
+
+    Assess network connectivity by checking connection status (via nmcli) in
+    sys-net.
+    """
+    command = b"nmcli networking connectivity check"
+
+    if not Util.get_qubes_version():
+        logger.error("QubesOS not detected, cannot check network.")
+        return False
+    try:
+        result = subprocess.check_output(["qvm-run", "sys-net", command])
+        return result is not None and result.decode("utf-8") == "full"
```

Comment:
The subprocess call on L205 will _always_ return an empty string (`b''`), because `qvm-run` doesn't return stdout/stderr from the domU unless `--pass-io` is given—so the function containing this check will always return `False`, as written. After ensuring stdio is properly handled, make sure to account for any whitespace; I expect `b'full\n'` will be the raw return value. You can match against that, or strip whitespace and compare the string.